### ISSUE TITLE
kvm ssh port 22 stuck

### ISSUE THREAD

**Issue by xchoice** [mistio/mist-ce#987]
I'm trying to connect using kvm to a libvirt, but no matter what I put in SSH Port, it still tries to connect using port 22.

```
2020-06-26 15:56:08,633 INFO  [mist.api.clouds.methods][uWSGIWorker8Core0] Adding new cloud in provider 'libvirt'
2020-06-26 15:56:08,643 WARNI [mist.api.helpers][uWSGIWorker8Core0] Got param 'machine_hostname' when expecting 'host', transforming.
2020-06-26 15:56:08,643 WARNI [mist.api.helpers][uWSGIWorker8Core0] Got param 'machine_name' when expecting 'alias', transforming.
2020-06-26 15:56:08,643 WARNI [mist.api.helpers][uWSGIWorker8Core0] Got param 'machine_user' when expecting 'username', transforming.
2020-06-26 15:56:08,643 WARNI [mist.api.helpers][uWSGIWorker8Core0] Got param 'machine_key' when expecting 'key', transforming.
2020-06-26 15:56:08,644 WARNI [mist.api.helpers][uWSGIWorker8Core0] Got param 'ssh_port' when expecting 'port', transforming.
2020-06-26 15:56:08,720 INFO  [mist.api.keys.base][uWSGIWorker8Core0] Associating key 6d12b72dd9be482986c443f48db0640d to machine 192-168-1-20
[pid: 51|app: 0|req: 2/8] 172.18.0.17 () {26 vars in 429 bytes} [Fri Jun 26 15:56:08 2020] GET /api/v1/keys => generated 285 bytes in 35 msecs (HTTP/1.1 200) 2 headers in 72 bytes (1 switches on core 0)
2020-06-26 15:56:09,250 INFO  [mist.api.keys.controllers][uWSGIWorker8Core0] Deploying key to host 192.168.1.20
2020-06-26 15:56:09,258 INFO  [mist.api.shell][uWSGIWorker8Core0] autoconfiguring Shell for machine 41efda5304b24a088c16159aabae0ee0:c1bbd6708a454aa19f4a8b9c60ec8745
2020-06-26 15:56:09,266 INFO  [mist.api.shell][uWSGIWorker8Core0] Got cloud & machine: 1 key associations
2020-06-26 15:56:09,267 INFO  [mist.api.shell][uWSGIWorker8Core0] Got keys 1
2020-06-26 15:56:09,268 INFO  [mist.api.shell][uWSGIWorker8Core0] Got users:['erika']
2020-06-26 15:56:09,268 INFO  [mist.api.shell][uWSGIWorker8Core0] Got ports:[22]
2020-06-26 15:56:09,268 INFO  [mist.api.shell][uWSGIWorker8Core0] ssh -i FileXvirt erika@192.168.1.20:22
2020-06-26 15:56:09,270 ERROR [mist.api.shell][uWSGIWorker8Core0] Got ssh error: NoValidConnectionsError(None, 'Unable to connect to port 22 on 192.168.1.20')
2020-06-26 15:56:09,270 ERROR [mist.api.shell][uWSGIWorker8Core0] Got ssh error: NoValidConnectionsError(None, 'Unable to connect to port 22 on 192.168.1.20')
2020-06-26 15:56:09,270 ERROR [mist.api.shell][uWSGIWorker8Core0] Got ssh error: NoValidConnectionsError(None, 'Unable to connect to port 22 on 192.168.1.20')
2020-06-26 15:56:09,272 ERROR [mist.api.clouds.controllers.main.controllers][uWSGIWorker8Core0] Could not connect to host 192.168.1.20.
2020-06-26 15:56:09,358 INFO  [mist.api.views][uWSGIWorker8Core0] MistError: MistError("Mist Error: Couldn't connect to host '192.168.1.20'.")
[pid: 68|app: 0|req: 2/9] 172.18.0.25 () {60 vars in 1205 bytes} [Fri Jun 26 15:56:08 2020] POST /api/v1/clouds => generated 52 bytes in 784 msecs (HTTP/1.0 500) 2 headers in 98 bytes (1 switches on core 0)
```
I did docker-compose down and up -d to see if that fixed it but it did not.

Also, why it's trying to deploy the key? I did that manually >.<

**Comment by xobotun** [comment]
I've encountered the same problem and was forced to do some ad-hockery.

I've downloaded the [shell.py](https://raw.githubusercontent.com/mistio/mist.api/master/src/mist/api/shell.py) via wget and replaced all `22` ports with `2222` strings with sed like `sed -i -e 's/22/2222/g' tmpfixes/shell.py`. This did not help me, though, as the script received the 22 ssh port from the somewhere outside and tried connecting to it first, and completely failed.

So I had to replace line 273 with more logic-breaking code: `'s/ports.append(2222)/ports = [2222]/g`. This one worked and I've finally added local KVM to the clouds. Still haven't tested if I could create VMs with it yet, as I'm new to KVM...

In order to attach "fixed" code to the container I appended a couple of lines to docker-compose, so the `api` service started looking like
```
  api:
    <<: *backend
    command: bin/wait-all bin/uwsgi
    expose:
      - 80
    volumes:
      - ./tmpfixes/shell.py:/mist.api/src/mist/api/shell.py
```
